sdks/java/harness/src/main/java/org/apache/beam/fn/harness/status/MemoryMonitor.java

```diff
@@ -573,6 +621,7 @@ private static File getLoggingDir() {
   public File dumpHeap()
       throws MalformedObjectNameException, InstanceNotFoundException, ReflectionException,
           MBeanException, IOException {
+    Preconditions.checkState(canDumpHeap);
```

Comment:
An error message here would be good, so if it ever happens the user will report it. Maybe something like "Bug! Attempt to dump heap even though it should be disabled."

Reply:
done